src/www/ui/api/Controllers/UploadController.php

```diff
@@ -837,4 +837,48 @@ public function getGroupsWithPermissions($request, $response, $args)
     $res["permGroups"] = $finalPermGroups;
     return $response->withJson($res, 200);
   }
+
+  /**
+   * Set the main license for the upload
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function setMainLicense($request, $response, $args)
+  {
+    try {
+      $uploadId = intval($args['id']);
+      $body = $this->getParsedBody($request);
+      $shortName = $body['shortName'];
+      $returnVal = null;
+      $license = null;
+      $licenseDao = $this->container->get('dao.license');
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadId)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if (empty($shortName)) {
+        $returnVal = new Info(400, "Short name missing from request.",
+          InfoType::ERROR);
+      } else {
+        $license = $licenseDao->getLicenseByShortName($shortName,
+          $this->restHelper->getGroupId());
+
+        if ($license === null) {
+          $returnVal = new Info(404, "License file with short name '$shortName' not found.",
+            InfoType::ERROR);
+        }
+      }
+      if ($returnVal !== null) {
+        return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+      }
+      $clearingDao = $this->container->get('dao.clearing');
+      $clearingDao->makeMainLicense($uploadId, $this->restHelper->getGroupId(), $license->getId());
```

Comment:
Please add a check to see if main license with the same group exists for the upload to prevent creating duplicates.

Reply:
done